pkg/micro-ecc/patches/0001-No-128-Bit-types-on-gcc.patch

```diff
@@ -0,0 +1,25 @@
+From ea5b0ba330ccf246317087e5eef4736c36b17158 Mon Sep 17 00:00:00 2001
+From: Leon Hielscher <Hielscher@fzi.de>
+Date: Mon, 8 Oct 2018 18:40:39 +0200
+Subject: [PATCH 4/4] No 128 Bit types on gcc.
+
+---
+ types.h | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/types.h b/types.h
+index 7cb1a28..0a0ff2d 100644
+--- a/types.h
++++ b/types.h
+@@ -51,7 +51,7 @@
+     #define uECC_WORD_SIZE 4
+ #endif
+ 
+-#if defined(__SIZEOF_INT128__) || ((__clang_major__ * 100 + __clang_minor__) >= 302)
++#if ((__clang_major__ * 100 + __clang_minor__) >= 302)
```

Comment:
that one I don't understand

Reply:
good question - @fzi-haxel?

Reply:
On x86_64, `__SIZEOF_INT128__` is defined, therefore `__int128` is used, but this leads to this GCC warning:
```
build/pkg/micro-ecc/types.h:98:18: error: ISO C does not support ‘__int128’ types [-Werror=pedantic]
   98 | typedef unsigned __int128 uECC_dword_t;
```

In retrospect, simply not using `__int128` does not seem to be the most elegant solution. Another option would be:
```
__extension__ typedef unsigned __int128 uECC_dword_t;
```

However, the GCC documentation points out the following when using `__extension__`
> only system header files should use these escape routes.

What do you think?

Reply:
I'm a bit confused: clang accepts `__int128` even though it is not part of the C standard but gcc does not?

What I also do not understand: if `__int128` is a built-in data type on gcc (>= 4.1) wouldn't I still get the warning even with the typedef? (Which definitiv would be actually used?)

Reply:
If `-Wpedantic` is enabled, yes.
Example: https://godbolt.org/z/bsYq49jGY

Reply:
So this typedef wouldn't help to get rid of the warning anyway.

Reply:
Using `__extension__` would get rid of the warning.

Reply:
Are you sure? I just tested it with (13.2.0) and didn't get this warning.

But GCC 13 now gives this warning during linking
```
/usr/bin/ld: warning: /work/tests/pkg/micro-ecc/bin/native/cpu/tramp.o: missing .note.GNU-stack section implies executable stack
```

Reply:
Ah, no, sorry, I think there was a PEBCAK issue (I always confuse the order of arguments for typedef).

Reply:
Opened an issue upstream with some suggested solutions